certificate-manager/src/test/java/io/strimzi/certs/OpenSslCertManagerTest.java

```diff
@@ -0,0 +1,171 @@
+        sbj.setSubjectAltNames(subjectAltNames);
+
+        testGenerateSelfSignedCert(key, cert, sbj);
+
+        key.delete();
+        cert.delete();
+    }
+
+    private void testGenerateSelfSignedCert(File key, File cert, Subject sbj) throws Exception {
+        ssl.generateSelfSignedCert(key, cert, sbj, 365);
+
+        Certificate c = certFactory.generateCertificate(new FileInputStream(cert));
+
+        c.verify(c.getPublicKey());
+
+        // subject verification if provided
+        if (sbj != null) {
+            if (c instanceof X509Certificate) {
+                X509Certificate x509Certificate = (X509Certificate) c;
+                Principal p = x509Certificate.getSubjectDN();
+
+                assertEquals(String.format("CN=%s, O=%s", sbj.commonName(), sbj.organizationName()), p.getName());
+
+                if (sbj.subjectAltNames() != null && sbj.subjectAltNames().size() > 0) {
+                    final Collection<List<?>> sans = x509Certificate.getSubjectAlternativeNames();
+                    assertNotNull(sans);
+                    assertEquals(sbj.subjectAltNames().size(), sans.size());
+                    for (final List<?> sanItem : sans) {
+                        assertTrue(sbj.subjectAltNames().containsValue(sanItem.get(1)));
+                    }
+                }
+            } else {
+                fail();
+            }
+        }
+    }
+
+    @Test
+    public void testGenerateSignedCert() throws Exception {
+
+        File caKey = File.createTempFile("tls", "key");
+        File caCert = File.createTempFile("tls", "crt");
+
+        ssl.generateSelfSignedCert(caKey, caCert, 365);
+
+        File key = File.createTempFile("tls", "key");
+        File csr = File.createTempFile("tls", "csr");
+        Subject sbj = new Subject();
+        sbj.setCommonName("MyCommonName");
+        sbj.setOrganizationName("MyOrganization");
+        Map<String, String> subjectAltNames = new HashMap<>();
+        subjectAltNames.put("DNS.1", "example1.com");
+        subjectAltNames.put("DNS.2", "example2.com");
+        sbj.setSubjectAltNames(subjectAltNames);
+
+        ssl.generateCsr(key, csr, sbj);
+
+        File cert = File.createTempFile("tls", "crt");
+        ssl.generateCert(csr, caKey, caCert, cert, 365);
+
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+        Certificate c = cf.generateCertificate(new FileInputStream(cert));
+        Certificate ca = cf.generateCertificate(new FileInputStream(caCert));
+
+        c.verify(ca.getPublicKey());
+
+        if (c instanceof X509Certificate) {
+            X509Certificate x509Certificate = (X509Certificate) c;
+            Principal p = x509Certificate.getSubjectDN();
+
+            assertEquals(String.format("CN=%s, O=%s", sbj.commonName(), sbj.organizationName()), p.getName());
+
+            // subject alternative names are not transferred automatically from a CSR to the final certificate
```

Comment:
I think I asked it already somewhere, but I'm not sure I got an answer.What does this mean for us? Does it mean that we need to disable hostname verification?

Reply:
For the current iteration yes. I guess it's the same problem EnMasse team has today, right? I opened an issue for investigating on how having SAN working support when using openssl.

Reply:
Ok, if the clients need to disable hostname verification, it probably should be mentioned in the docu (ideally with a record in the properties example). There was a KIP which enabled it by default which I guess ended in Kafka 2.0 - so users will need to disable it. We should also hjave an issue for it - because this is quite an important problem.